esx_service/vmdk_ops.py

```diff
@@ -660,11 +660,16 @@ def listVMDK(tenant):
             for x in vmdks]
 
 
-# Return VM managed object, reconnect if needed. Throws if connection fails twice.
-# returns None if the uuid is not found
-def findVmByUuid(vm_uuid):
+
+def findVmByUuid(vm_uuid, is_vc_uuid=False):
+    """
+    Find VM by vm_uuid.
+    is_vc_uuid should be true if vm_uuid is vc uuid, else it should be false.
+    Return VM managed object, reconnect if needed. Throws if connection fails twice.
+    Returns None if the uuid is not found
+    """
     si = get_si()
-    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
+    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, is_vc_uuid)
```

Comment:
what if instanceUuid of one VM is the same as BIOS UUID of another one ? slim chances, but possible, is it not ?